### ISSUE TITLE
loss increased in 1D wave equation

### ISSUE THREAD

**Issue by yujingyan222** [levimcclenny/SA-PINNs#5]
Dear @levimcclenny ,
Thanks for the innovative contribution. I tried to run the wave equation 1D but the loss increased all the time. I don't know why. Could you please help me check it? Thanks a lot!!
![image](https://user-images.githubusercontent.com/107137942/173843082-3820e3b1-1aba-4ca2-a6bc-95c5aedc47e4.png)
![image](https://user-images.githubusercontent.com/107137942/173843220-a826acda-b867-4de2-a774-3d2aa53ed370.png)

**Comment by levimcclenny** [comment]
Hey @yujingyan222, Thanks for the comment. This can occur when the SA-PINN is searching the loss space and performing the min-max optimization. This is a unique characteristic of the training as losses that are monotonically decreasing, especially in PINNs, can result in poor solutions. The AC results that are presented in the paper have this same phenomenon on the total loss value. 

From your code snippet it looks correct, just continue the training. 

The key is not to look at just the total loss (since that is modified by the weight values as they are trained) but to look at the MSE on the individual parameters with the loss weights removed.

**Comment by levimcclenny** [comment]
Also, have a look at your learning rate schedules. The default parameters are unique to the problems at hand (i.e. AC, Helmholtz, etc) and the learning rates on your ADAM optimizer can heavily influence the training.

**Comment by yujingyan222** [comment]
> Also, have a look at your learning rate schedules. The default parameters are unique to the problems at hand (i.e. AC, Helmholtz, etc) and the learning rates on your ADAM optimizer can heavily influence the training.

Thanks a lot, the result is ok now. Thanks again.
